Compilation failure with feature `candle-cuda`

**Describe the bug**
The crate fails to compile when enabling the `candle-cuda` feature.

**To Reproduce**
Add the following to your Cargo.toml:
`burn = { version = "0.20.0-pre.5", features = ["candle-cuda"] }`

**Expected behavior**
The crate should compile with this feature enabled.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: Linux (NixOS Unstable)
 - Version: 0.20.0-pre.5 and latest commit on master

**Additional context**
Compiler error:
```
error[E0252]: the name `BackTrace` is defined multiple times
 --> /home/andreas/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/burn-candle-0.20.0-pre.5/src/backend.rs:6:5
  |
4 | use burn_std::backtrace::BackTrace;
  |     ------------------------------ previous import of the type `BackTrace` here
5 | use burn_std::{
6 |     backtrace::BackTrace,
  |     ^^^^^^^^^^^^^^^^^^^^-
  |     |
  |     `BackTrace` reimported here
  |     help: remove unnecessary import
  |
  = note: `BackTrace` must be defined only once in the type namespace of this module

error[E0559]: variant `ExecutionError::Generic` has no field named `context`
   --> /home/andreas/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/burn-candle-0.20.0-pre.5/src/backend.rs:248:25
    |
248 |                         context: format!("Can't sync the cuda device: {err}"),
    |                         ^^^^^^^ `ExecutionError::Generic` does not have this field
    |
    = note: available fields are: `reason`

Some errors have detailed explanations: E0252, E0559.
For more information about an error, try `rustc --explain E0252`.
error: could not compile `burn-candle` (lib) due to 2 previous errors
```

--- a/crates/burn-candle/src/backend.rs
+++ b/crates/burn-candle/src/backend.rs
@@ -1,7 +1,5 @@
 use std::marker::PhantomData;
 
-#[cfg(feature = "cuda")]
-use burn_std::backtrace::BackTrace;
 use burn_std::{
     backtrace::BackTrace,
     rand::{SeedableRng, StdRng},
@@ -245,7 +243,7 @@ impl<F: FloatCandleElement, I: IntCandleElement> Backend for Candle<F, I> {
                 device
                     .synchronize()
                     .map_err(|err| ExecutionError::Generic {
-                        context: format!("Can't sync the cuda device: {err}"),
+                        reason: format!("Can't sync the cuda device: {err}"),
                         backtrace: BackTrace::capture(),
                     })?;
             }

